test/utilities.go

```diff
@@ -0,0 +1,196 @@
+package test
+
+import (
+	"bufio"
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/nsf/jsondiff"
+)
+
+// EqualJson compares two json strings.
+// returns true if considered equal, false otherwise.
+// The error returns the difference.
+// For reference: j1 is the baseline, j2 is the test
+func EqualJson(j1, j2 string) (ans bool, err error) {
+
+	// Sorting the elements inside json arrays is important to avoid false
+	// negatives due to element order mismatch.
+	j1, err = SortJsonArrays(j1)
+	if err != nil {
+		return false, err
+	}
+
+	j2, err = SortJsonArrays(j2)
+	if err != nil {
+		return false, err
+	}
+
+	// This func uses jsondiff.
+	// options configures it.
+	options := jsondiff.Options{
+		Added:            jsondiff.Tag{Begin: "___ADDED___", End: ""},
+		Removed:          jsondiff.Tag{Begin: "___REMED___", End: ""},
+		Changed:          jsondiff.Tag{Begin: "___DIFFER___", End: ""},
+		ChangedSeparator: " -> ",
+	}
+	options.PrintTypes = false
+	d, str := jsondiff.Compare([]byte(j1), []byte(j2), &options)
+	// If fully match, return true
+	if d == jsondiff.FullMatch {
+		return true, nil
+	}
+
+	// Check the difference, and decide if it is still accepted as "equal"
+	// Scan line by line, and examine the difference
+	scanner := bufio.NewScanner(strings.NewReader(str))
+	for scanner.Scan() {
+		line := scanner.Text()
+		// If baseline has the property (j1), but the test (j2) has it removed
+		if strings.Contains(line, "___REMED___") {
+			// If the value of the property is false, then accept it
+			// The default value is false. Missing property is considered as false
+			propValue := strings.SplitN(line, ": ", 2)
```

Comment:
Could you give an example of what line would be? Specifically the `len(propValue) == 1` case.

Reply:
___REMED___},

Reply:
So `{ "field":"value"` compared to `{"field":"value"}` would be considered equal?